packages/contracts-bridge/contracts/BridgeMessage.sol

```diff
@@ -326,6 +363,35 @@ library BridgeMessage {
         return _message.slice(0, TOKEN_ID_LEN, uint40(Types.TokenId));
     }
 
+    /**
+     * @notice Retrieves the hook contract EVM address from a TransferWithHook
+     * @param _transferAction The message
+     * @return The hook contract address as bytes32
+     */
+    function evmHook(bytes29 _transferAction)
+        internal
+        pure
+        typeAssert(_transferAction, Types.TransferToHook)
+        returns (address)
+    {
+        return _transferAction.indexAddress(13);
+    }
+
+    function extraData(bytes29 _transferAction)
+        internal
+        pure
+        typeAssert(_transferAction, Types.TransferToHook)
+        returns (bytes29)
+    {
+        // anything past the end is the extradata
```

Comment:
thought: what if the action is equal to `MIN_TRANSFER_LEN`? does `slice` behave as expected if the number of bytes to slice is 0

Reply:
```
function slice(bytes29 memView, uint256 _index, uint256 _len, uint40 newType) internal pure returns (bytes29) {
        uint256 _loc = loc(memView);

        // Ensure it doesn't overrun the view
        if (_loc + _index + _len > end(memView)) {
            return NULL;
        }

        _loc = _loc + _index;
        return build(newType, _loc, _len);
    }
```

it'll return a slice of 0 length